src/HTTP/Throttling/Increment/MultiplierIncrement.php

```diff
@@ -0,0 +1,38 @@
+<?php declare(strict_types=1);
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ ********************************************************************
+ */
+
+namespace ILIAS\HTTP\Throttling\Increment;
+
+/**
+ * @author Thibeau Fuhrer <thibeau@sr.solutions>
+ */
+class MultiplierIncrement implements DelayIncrement
+{
+    protected float $multiplier;
+
+    public function __construct(float $multiplier)
+    {
+        $this->multiplier = $multiplier;
+    }
+
+    public function increment(float $delay) : float
+    {
+        return ($this->multiplier * $delay);
```

Comment:
Is there an upper limit to the delay? I am a bit concerned that an attacker might abuse this feature for a DoS attack: When all worker threads are sleeping (potentially for a long time), there might not be any available for legitimate requests. Or is this already considered elsewhere?

Reply:
You're right, there isn't an upper limit yet but we'll have to take this into account. Thanks for the feedback!